File: packages/flow_widget_ios/ios/flow_widget_ios/Sources/flow_widget_ios/FlowWidgetCapabilitiesProvider.swift
import Foundation

enum FlowWidgetCapabilitiesProvider {
    static func capabilities() -> [String: Bool] {
        var flags: [String: Bool] = [
            "homeWidgets": true,
            "lockScreenWidgets": true,
            "interactiveWidgets": true,
            "configurableWidgets": true,
            "timelineProviders": true,
            "liveActivities": liveActivitiesSupported,
            "dynamicIsland": dynamicIslandSupported,
            "pinWidget": false,
            "backgroundUpdates": true,
            "scheduledUpdates": true,
            "pushUpdates": true,
            "remoteImageCaching": true,
            "appGroups": true,
            "wearTiles": false,
            "complications": false,
            "multipleInstances": true,
            "resizing": true,
            "themeSynchronization": true,
            "appIntents": true,
        ]
        return flags
    }

    private static var liveActivitiesSupported: Bool {
        if #available(iOS 16.2, *) {
            return true
        }
        return false
    }

    private static var dynamicIslandSupported: Bool {
        if #available(iOS 16.2, *) {
            return true
        }
        return false
    }
}

File: packages/flow_widget_android/android/src/main/kotlin/dev/flowwidget/android/FlowWidgetCapabilitiesProvider.kt
package dev.flowwidget.android

import android.os.Build

/**
 * Reports Android-specific capability flags for the flow_widget Dart API.
 */
object FlowWidgetCapabilitiesProvider {
    fun getCapabilities(): Map<String, Boolean> {
        val supportsPin = Build.VERSION.SDK_INT >= Build.VERSION_CODES.O
        val supportsInteractive = Build.VERSION.SDK_INT >= Build.VERSION_CODES.S

        return mapOf(
            "homeWidgets" to true,
            "lockScreenWidgets" to (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU),
            "interactiveWidgets" to supportsInteractive,
            "configurableWidgets" to true,
            "timelineProviders" to true,
            "liveActivities" to false,
            "dynamicIsland" to false,
            "pinWidget" to supportsPin,
            "backgroundUpdates" to true,
            "scheduledUpdates" to true,
            "pushUpdates" to false,
            "remoteImageCaching" to true,
            "appGroups" to true,
            "wearTiles" to false,
            "complications" to false,
            "multipleInstances" to true,
            "resizing" to (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S),
            "themeSynchronization" to (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S),
            "appIntents" to false,
        )
    }
}
